src/components/node-options-form/index.tsx

```diff
@@ -0,0 +1,61 @@
+import useFlowStore, { flowSelector } from "@pb/store/flow-builder.store";
+import React, { useCallback, useEffect, useState } from "react";
+import { shallow } from "zustand/shallow";
+
+import BaseInput from "../base-input";
+import type { PBNodeProps } from "../flow-builder/flow-builder.types";
+import InfoCard from "../info-card";
+import { type UpdateNodeConfig } from "./node-options-form.types";
+
+const NodeOptionsForm = ({ id, data }: PBNodeProps) => {
+  const { updateNodeConfigs } = useFlowStore(flowSelector, shallow);
+
+  const { description, icon, label, nodeConfigs: initialNodeConfigs } = data;
+  const [nodeConfigs, setNodeConfigs] = useState(initialNodeConfigs);
+
+  useEffect(() => {
+    setNodeConfigs(initialNodeConfigs);
+  }, [initialNodeConfigs]);
+
+  const updateNodeConfig: UpdateNodeConfig = useCallback(
+    (key) => (event) => {
+      setNodeConfigs((prevNodeConfigs) => ({
+        ...(prevNodeConfigs || {}),
+        [key]: event.target.value,
+      }));
+      updateNodeConfigs(id, key, event.target.value || "");
+    },
+    [id, updateNodeConfigs]
+  );
+
+  return (
+    <div className="h-full max-w-[32rem] flex-1 bg-base-100 px-4 py-9">
+      <InfoCard
+        containerCustomClasses="pb-6 items-start"
+        description={description}
+        icon={icon}
+        title={label}
+      />
+      <BaseInput
+        key={`message-${id}`}
+        containerCustomClasses="pt-6 pb-3"
+        inputType="textarea"
+        value={nodeConfigs?.message}
+        label="Message"
+        placeholder="Type in the question here..."
+        onChange={updateNodeConfig("message")}
+      />
+      <BaseInput
+        key={`answer-${id}`}
+        containerCustomClasses="pt-6 pb-3"
+        inputType="textarea"
+        label="Expected Answer"
+        placeholder="Leave it empty if nothing is expected from the user."
+        value={nodeConfigs?.expectedAnswer}
+        onChange={updateNodeConfig("expectedAnswer")}
+      />
+    </div>
+  );
+};
+
+export default NodeOptionsForm;
```

Comment:
> When you provide your input and navigate to another node (step), refreshing the page will remove the current state, causing any previously entered values to be cleared! Consequently, the input fields will revert to their default values, without retaining them as placeholders. In other words, the inputs will discard the entered values and display the original default values as actual values, rather than placeholders.

This PR covers only the UI elements, Nothing is saved on the browser level now, we only have the state, when it's connected with the backend we will consider the change

Reply:
> When you add a new node and enter the values, and then proceed to add another node without assigning any values to it, try navigating between them. You will notice that the values from the first node are being copied to the second node.

confirmed, will address this

Reply:
> The new node will always adopt the value of the previous node, and vice versa. This process sometimes results in a rapid renderer switch between default values and actual values, leading to multiple unnecessary state updates.

can you please clarify more on this point